### PR TITLE
feat: Add user search to GraphQL API

### LINKED ISSUE #175: Implement fuzzy searching for users api
### Problem/Why
As the user list grows, an application admin should be able to search for a user in the list.

### Goal/What
The application admin should be able to filter the users list via a fuzzy search for email, name, and username.

### Solution/How
Update the GraphQL endpoint to allow a search parameter to be added which implements filtering users.  For inspiration: https://dev.to/khalyomede/create-an-easy-fuzzy-loose-search-with-laravel-5-34a4  

### To Do / Acceptance Criteria

- [x] Update the users list endpoint to accept a search parameter
- [x] Implement fuzzy search logic on username, name, and email

### Review Checklist

- [ ] API returns filtered user results

### PR DESCRIPTION
**Pull request by gmeben**
This extends the GraphQL API to search for users by various search terms.

Closes #175 

Example GraphQL Playground query:

```graphql
query SearchUsers {
  userSearch(term:"abcdef", first: 10, page: 1) {
    data {
      id	
      name
      email
      username
    }
  }
}
```

Edit: changed the name of the function argument from 'search' to 'term'

### CODE REVIEW

File: backend/graphql/schema.graphql [hunk 1]
```diff
@@ -21,7 +21,11 @@ type Query {
     "Return pre-defined user roles in the application"
     role(id: ID @eq): Role @find
 
+    "Return a permission by ID"
     permission(id: ID @eq): Permission @find
+
```
Comment (wreality) [inline backend/graphql/schema.graphql:26]: Should this be a separate query or should the search parameter be added to the existing users query?

File: backend/config/scout.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    | Here you may configure your Algolia settings. Algolia is a cloud hosted
+    | search engine which works great with Scout out of the box. Just plug
+    | in your application ID and admin API key to get started searching.
+    |
+    */
+
+    'algolia' => [
+        'id' => env('ALGOLIA_APP_ID', ''),
+        'secret' => env('ALGOLIA_SECRET', ''),
+    ],
```
Comment (wreality) [inline backend/config/scout.php:117]: Does this depend on an algolia API key to operate?  Excuse my lack of knowledge wrt to scout.

**Review by wreality — COMMENTED** [review]
Just a few questions to make sure I understand what's going on here.

File: backend/graphql/schema.graphql (hunk 1 above)
Comment (gmeben) [reply backend/graphql/schema.graphql:26]: Good question. I suppose if I had the option, I would add it to the existing users query, but I have not looked into how to combine two different directives (`eq` and `@search`) on the same query.

File: backend/config/scout.php (hunk 2 above)
Comment (gmeben) [reply backend/config/scout.php:117]: Nope, this Algolia section is leftover from the default config file. I can remove this section if you like.

File: backend/graphql/schema.graphql (hunk 1 above)
Comment (wreality) [reply backend/graphql/schema.graphql:26]: users, not user

I think searchUsers is probably a better name than users. Perhaps this should just replace the users endpoint.

File: backend/config/scout.php (hunk 2 above)
Comment (wreality) [reply backend/config/scout.php:117]: Nope that's fine, just wanted to make sure I understood what was going on.

File: backend/graphql/schema.graphql (hunk 1 above)
Comment (gmeben) [reply backend/graphql/schema.graphql:26]: That's a good suggestion. I'll refactor.

**Review by wreality — APPROVED** [review]

**Pull request merged**